Add 0.14.0 CHANGELOG (#3903)
holoviz/panel: CHANGELOG.md

# Releases

## Version 0.14.0

Date: 2022-09-30

> This release focuses on three main themes, support for running Panel entirely in the browser using WebAssembly (via Pyodide and PyScript), quality of life improvements to make it easier to build responsive and performant applications and finally improvements to static typing and docstrings to enhance the developer experience.

Reviewer: ```suggestion
This release focuses on three main themes: support for running Panel apps entirely in the browser using WebAssembly (via Pyodide and PyScript), improvements in the app-user experience by making it easier to build responsive and performant applications, and finally improvements in the developer experience through static typing and docstrings.
```

(I'm making a guess that by "quality of life" you mean "for the user of a Panel app", and that you are trying to distinguish between improvements that are visible to an app visitor and improvements for app developers. If not, please reword.)

> This release focuses on three main themes, support for running Panel entirely in the browser using WebAssembly (via Pyodide and PyScript), quality of life improvements to make it easier to build responsive and performant applications and finally improvements to static typing and docstrings to enhance the developer experience.

Reviewer 2: Please convert 3 themes to a list. It each much more appealing and easy to read.

> 

Reviewer: These changes represent a major step forward in the Panel user and developer experience and the ability to share Panel apps widely without the need for maintaining a central server.

Many, many thanks to everyone who filed issues or contributed to this release. In particular we would like to thank @janimo, @xavArtley, @thuydotm, @jmosbacher, @dmarx, 2WoLpH, @ipopa144, @sdc50 and the core team consisting of @philippjfr, @Hoxbro, @maximlt and @MarcSkovMadsen.

### Features

- Add support for converting Panel apps to pyscript/pyodide (#3817, #3830, #3851, #3856, #3857, #3858, #3860, #3861, #3863, #3864, #3868, #3878)
- Manage our own CDN (#3867, #3870)

> - Add `pn.widget` helper function (#1826)

Reviewer 3: ```suggestion
- Add `pn.widgets` helper function (#1826)
```

Author: pn.widget is correct.

Reviewer 3: Ah, sorry.

> - Add `pn.widget` helper function (#1826)

Reviewer: ```suggestion
- Add `pn.widget` helper function (#3589, #1826)
```
(1826 is just the issue, not the PR, though including it does make sense because the PR doesn't explain what it's for.)

Has this subsequently been documented? If so the doc PR should also be linked here. If not, need to document it!

- Add ability to `defer_load` of components (#3882)
- Add `config.exception_handler` to easily capture, log and notify users about errors (#3893)
- Implement `pn.cache` function for memoization support (#2411)

> - Rewrite server extension to run Panel applications in kernels (#3763)

Reviewer: ```suggestion
- Rewrite server extension to run Panel applications in kernels so that previews run in the same environment as the deployed app (#3763)
```

Reviewer 2: so that previews run in the same environment as the notebook and code....

- Add liveness endpoint (#3832)
- Add ability to define authorization callback (#3777)
- Add memray profiler (#3509)

### Enhancements

- Ensure OAuth redirects to requested app and retains query arguments (#3555)
- Add extension entry point (#3738)
- Update Admin Logs page to use `Tabulator` (#3694)
- Ensure `location.unsync` unsets query params (#3806)

> - Allow none value on numeric sliders and `LiteralInput` (#3174)

Reviewer: ```suggestion
- Allow None value on numeric sliders and `LiteralInput` (#3174)
```

- Allow serving admin panel with `pn.serve` (#3798)
- Improve `ReactiveHTML` loop support and validation (#3813)
- Support declaring `Perspective.plugin_config` pane (#3814)
- Do not flicker busy indicator during `--autoreload` check (#3804)
- Improve robustness of `state.curdoc` in threaded and async contexts (#3776, #3810, #3834)
- Supporting datetime bounds for `DatetimePicker` and `DatetimeRangePicker` (#3788)
- Allow setting the Oauth provider using environment variables (#3698)
- Implement `Player.value_throttled` (#3756)
- Ensure that URL query parameters are preserved during OAuth (#3656)
- Improve `Markdown` code syntax highlighting (#3758)
- Ensure components do not re-render if `background` or `loading` parameters change (#3599)
- Add ability to define admin dashboard plugins (#3668)
- Do not calculate embed state for disabled widgets (#3757)
- Add hard bounds to editable sliders (#3739)

### Documentation

- Overhaul documentation (#3568)

> - Improves the Fast Template docstrings (#3570)

Reviewer: ```suggestion
- Improve Fast Template docstrings (#3570)
```

> - Reorganize docs and convert static notebooks to markdown (#3875, #3833)

Reviewer: ```suggestion
- Reorganize docs and convert static notebooks to Markdown (#3875, #3833)
```

> - Added DevelopingCustomModels to the webpage (#3710)

Reviewer: ```suggestion
- Add DevelopingCustomModels to the webpage (#3710)
```

- Improve typing (#3561, #3562, #3592, #3604, #3714, #3729)

### Compatibility & Version updates

- Support ipywidgets>=8.0 (#3782)
- Bump jsoneditor package (#3838)
- Upgrade to Tabulator 5.3.2 (#3784)
- Improve Django compatibility (#3843, #3835)

> - Removed all usage of deprecated `Pane`

Reviewer 3: This is missing a #

> - Removed all usage of deprecated `Pane`

Reviewer: ```suggestion
- Remove all usage of deprecated `Pane`
```

### Bugs

#### Server

> - Ensure closed websocket does not cause errors

Reviewer 3: missing #

- Handle session and websocket close cleanly (#3769)
- Fix prefix handling for admin page (#3809)
- Support admin dashboard in multi-process deployments (#3812)

> - Improved document cleanup when not invoked using `server_destroy` (#3842)

Reviewer: ```suggestion
- Improve document cleanup when not invoked using `server_destroy` (#3842)
```

- Ensure `pn.state.execute` dispatches immediately if possible (#3859)
- Ensure autoload.js resources are appropriately prefixed (#3873)

#### Notebook

- Fix support for copying cells and creating new views in JupyterLab (#3652)
- Ensure output renders in VSCode notebook with latest ipywidgets (#3765)

> - Resolved issues with Jupyter slowdown due to event_loop patching on Windows (#3770)

Reviewer: ```suggestion
- Resolve issues with Jupyter slowdown due to event_loop patching on Windows (#3770)
```

- Update `page` Parameter when pagination is 'local' (#3704)
- Do not apply `sorters` on `Tabulator` cell edits (#3744)
- Ensure `Tabulator.controls` renders (#3768)
- Ensure correctness of event row and selection indices in `Tabulator` (#3771, #3841)
- Fix issues with frontend and backend sorters being out of sync in Tabulator (#3825, #3839)
- Fix default values of a list header filter in `Tabulator` (#3826)
- Fix the edit event with a python filter in `Tabulator` (#3829)
- Disable client-side date filtering on `Tabulator` (#3849)
- Support editing of pandas masked array dtypes in `Tabulator` (#3850)
- Fix issues editing a cell when client-side filtering applied (#3852)
- Do not recompute data when local pagination is enabled (#3854)
- Don't skip filtering when the column name is undefined (#3862)

#### Misc

- Fix `FileInput.save` (#3579)
- Fix issues with `Matplotlib.high_dpi` option (#3591, #3594)
- Ensure layout recomputes on `HTML`/`Markdown` re-rerender (#3616)
- Allow overriding all widget parameters on `Param` pane (#3754)
- Ensure `DatePicker` start/end are transformed when jslinked (#3759)
- Ensure notifications can be enabled without a template (#3820)

> - Ensure `ReactiveHTML` inline callback on loop variables return correct node (#3840)

Reviewer: ```suggestion
- Ensure `ReactiveHTML` inline callbacks on loop variables return correct node (#3840)
```
? (Not sure what the right fix is, but something doesn't agree!)

# Releases

## Version 0.14.0

Date: 2022-09-30

> This release focuses on two main themes, support for running Panel entirely in the browser using WebAssembly (via Pyodide and PyScript), quality of life improvements to make it easier to build responsive and performant applications and finally improvements to static typing and docstrings to enhance the developer experience.

Reviewer 3: ```suggestion
This release focuses on three main themes, support for running Panel entirely in the browser using WebAssembly (via Pyodide and PyScript), quality of life improvements to make it easier to build responsive and performant applications and finally improvements to static typing and docstrings to enhance the developer experience.
```

Isn't this three things?
1) support for running Panel entirely in the browser using WebAssembly (via Pyodide and PyScript
2) quality of life improvements to make it easier to build responsive and performant applications
3) improvements to static typing and docstrings to enhance the developer experience
